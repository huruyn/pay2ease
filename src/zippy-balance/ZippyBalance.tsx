import { useMsal } from '@azure/msal-react';
import { useEffect, useState } from 'react';
import NumberFormat from 'react-number-format';
import { Button, Spinner } from 'react-bootstrap';
import { useNavigate } from 'react-router-dom';
import Api, { Account } from '../api';
import { useAppSelector } from '../app/hooks';
import { selectUser } from '../features/user/userSlice';
import PageContainer from '../common/PageContainer';

export const ZippyBalancePure = ({ account }: { account: Account | undefined }) => {
  const navigate = useNavigate();

  return (
    <>
      <PageContainer title="Zippy Balance" subTitle="Welcome back!" backdropImage="backdrop-image-1">
        <div className="zippy-balance">
          { account !== undefined ? (
            <>
              <div className="account-balance">
                <div>Your Zippy Balance is</div>
                <p>
                  <NumberFormat
                    className="ms-3"
                    value={account.balance}
                    displayType="text"
                    prefix="$ "
                    suffix=" CAD"
                    thousandSeparator
                    decimalScale={2}
                    fixedDecimalScale
                  />
                </p>
              </div>
              <div className="account-name">{account.name}</div>
            </>
          ) : (
            <Spinner className="spinner" animation="border" role="status">
              <span className="visually-hidden">Loading...</span>
            </Spinner>
          )}
        </div>
        { account !== undefined && (
          <div className="action pair">
            <Button
              className="zippy-btn"
              onClick={() => navigate('/my-wallet/zippy-money/send/transaction-start')}
            >
              Send Money
            </Button>
            <Button
              className="zippy-btn"
              onClick={() => navigate('/my-wallet/zippy-money/request/transaction-start')}
            >
              Request Money
            </Button>
          </div>
        )}
      </PageContainer>
    </>
  );
};

export default function ZippyBalance() {
  const { instance, accounts: msalAccounts } = useMsal();
  const [account, setAccount] = useState<Account | undefined>();
  const { user } = useAppSelector(selectUser);

  useEffect(() => {
    if (user) {
      new Api(instance, msalAccounts[0])
        .listAccounts()
        .then((accounts) => {
          setAccount(accounts[0]);
        });
    }
  }, [msalAccounts, user]);

  return <ZippyBalancePure account={account} />;
}
