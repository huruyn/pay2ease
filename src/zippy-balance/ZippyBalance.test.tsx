import { BrowserRouter } from 'react-router-dom';
import { ZippyBalancePure } from './ZippyBalance';

const ReactTestRenderer = require('react-test-renderer');

describe('ZippyBalance Component', () => {
  it('matches the snapshot when loaded', () => {
    const tree = ReactTestRenderer.create(
      <BrowserRouter>
        <ZippyBalancePure account={{ id: 1, name: '8000 001 000000000', balance: 1000 }} />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches the snapshot when loading', () => {
    const tree = ReactTestRenderer.create(
      <BrowserRouter>
        <ZippyBalancePure account={undefined} />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
