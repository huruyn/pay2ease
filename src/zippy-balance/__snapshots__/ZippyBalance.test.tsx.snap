// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZippyBalance Component matches the snapshot when loaded 1`] = `
<div
  className=""
>
  <div
    className="zippy-cash-container-wrap"
  >
    <div
      className="backdrop backdrop-image-1"
    />
    <div
      className="zippy-cash-container"
    >
      <div
        className="zippy-cash-container-header"
      >
        <h1
          className="title"
        >
          Zippy Balance
        </h1>
        <h2
          className="catch"
        >
          Welcome back!
        </h2>
      </div>
      <div
        className="zippy-cash-container-body"
      >
        <div
          className="zippy-balance"
        >
          <div
            className="account-balance"
          >
            <div>
              Your Zippy Balance is
            </div>
            <p>
              <span
                className="ms-3"
              >
                $ 1,000.00 CAD
              </span>
            </p>
          </div>
          <div
            className="account-name"
          >
            8000 001 000000000
          </div>
        </div>
        <div
          className="action pair"
        >
          <button
            className="zippy-btn btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Send Money
          </button>
          <button
            className="zippy-btn btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Request Money
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ZippyBalance Component matches the snapshot when loading 1`] = `
<div
  className=""
>
  <div
    className="zippy-cash-container-wrap"
  >
    <div
      className="backdrop backdrop-image-1"
    />
    <div
      className="zippy-cash-container"
    >
      <div
        className="zippy-cash-container-header"
      >
        <h1
          className="title"
        >
          Zippy Balance
        </h1>
        <h2
          className="catch"
        >
          Welcome back!
        </h2>
      </div>
      <div
        className="zippy-cash-container-body"
      >
        <div
          className="zippy-balance"
        >
          <div
            className="spinner spinner-border"
            role="status"
          >
            <span
              className="visually-hidden"
            >
              Loading...
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
